src/Native/LdaNative/model_block.cpp

```diff
@@ -324,14 +337,20 @@ namespace lda
             alias_offset += alias_row_size;
         }
 
-        mem_block_size_ = dict_[num_vocabs_ - 1].end_offset_;
+		uint64_t size = dict_[num_vocabs_ - 1].end_offset_;
+		if (size > numeric_limits<size_t>::max())
+			throw bad_alloc();
```

Comment:
I see sometimes you `throw bad_alloc()`, but I've seen 2 other places where you just call `bad_alloc()` without throwing. Why the difference?

Reply:
bug, need to throw

---
In reply to: [276749841](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276749841) [](ancestors = 276749841)